### DISCUSSION TITLE
new keyword ??

### DISCUSSION THREAD

**Discussion by kdub-nz** [wtertinek/Linq2Acad#43] [General]
From Part 7 Blog post.

It's somewhat comforting to know I'm not the only one who does this :)
![2024-02-19_12-07-52](https://github.com/wtertinek/Linq2Acad/assets/2830167/9602509c-f54a-4990-9b22-04abeb7da9ed)

Regards,

**Comment by kdub-nz** [comment]
On a less irreverant note :
Regarding Listing 9

`public class GeneralHelper : IDisposable
{
  private Database db;
  private Transaction tr;
 
  public GeneralHelper()
  {
    db = Application.DocumentManager.MdiActiveDocument.Database;
    tr = db.TransactionManager.StartTransaction();
  }`


I had expected to see the field names to have an underscore prefix, as you've done elsewhere.

ie:
`private Database _db;
  private Transaction _tr;`
this makes scanning the code more intuitive ( well, for me anyway :) )

Regards,

**Reply by wtertinek** [reply]
😄 The code in the blog posts hasn't found its way through the compiler, so it is possible that I invented a few more new keywords. 😉

Concerning the prefix it's actually the same with the newly invented keywords: the blog posts developed over the course of a year or so, I didn't fully check the coding style. But I hope the code still makes sense to you. 😉

In the Linq2Acad codebase there are no prefixes for member variables. But it is more a thing of personal preference. In my opinion, it should make no difference to the understanding of the code whether a variable belongs to a method or the class.

**Comment by kdub-nz** [comment]
Thanks Wolfgang
